runtime/parachains/src/util.rs

```diff
@@ -39,3 +40,20 @@ pub fn make_persisted_validation_data<T: paras::Config + hrmp::Config>(
 		max_pov_size: config.max_pov_size,
 	})
 }
+
+/// Take the active subset of a set containing all validators.
+pub fn take_active_subset<T: Clone>(active_validators: &[ValidatorIndex], set: &[T]) -> Vec<T> {
+	let subset: Vec<_> = active_validators.iter()
+		.filter_map(|i| set.get(i.0 as usize))
+		.cloned()
+		.collect();
+
+	if subset.len() != active_validators.len() {
```

Comment:
the size doesn't have to match, right? what we want to check is that `subset.iter().max().map_or(false, |m| m >= active_validators.len())`

Reply:
Ah, no, sorry, I confused a few names. Yes, it should be alright.

Reply:
Isn't it the same thing? If any index is out of bounds, the subset will be missing entries due to the `filter_map` above.